### ISSUE TITLE
Semantic3D Writes Empty Point Clouds

### ISSUE THREAD

**Issue by renwang435** [aboulch/ConvPoint#14]
The model successfully trains and tests, but fails to write the benchmark results for assessment.

In generating the prediction files, I run `python semantic3d_benchmark_gen.py --testdir path_to_original_test_data --savedir /path_to_save_dir_benchmark --refdata path_to_data_processed --reflabel path_to_prediction_dir` as required, and the output seems to indicate completion:

> Getting labeled points...
> Done 439458 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1413340 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 722670 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 564620 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 616367 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 554140 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 589729 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1133226 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1372009 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 485595 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 2300733 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1245062 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1859927 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1363112 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> Getting labeled points...
> Done 1115522 points
> Create KDTree...
> Done
> Iteration on the original point cloud...
> birdfountain_station1_xyz_intensity_rgb
> castleblatten_station1_intensity_rgb
> castleblatten_station5_xyz_intensity_rgb
> marketplacefeldkirch_station1_intensity_rgb
> marketplacefeldkirch_station4_intensity_rgb
> marketplacefeldkirch_station7_intensity_rgb
> sg27_station10_intensity_rgb
> sg27_station3_intensity_rgb
> sg27_station6_intensity_rgb
> sg27_station8_intensity_rgb
> sg28_station2_intensity_rgb
> sg28_station5_xyz_intensity_rgb
> stgallencathedral_station1_intensity_rgb
> stgallencathedral_station3_intensity_rgb
> stgallencathedral_station6_intensity_rgb

But when looking at the labels for each and every one of the test point clouds, they are **empty.** It is impossible to assess the correctness of the code as the function is not examinable. What is the reason are no written labels for the test files?

**Comment by yaoyao1900** [comment]
Hallo, I have the same problem as you.
semantic3d_seg.py 
`python semantic3d_benchmark_gen.py  --rootdir, path to processed test npy file, --savedir, path to the state_dict.pth file, --test`
This Test process takes a long time and I have got a new test npy file. The benchmark in this npy file is all 0.
semantic3d_benchmark_gen.py
`python semantic3d_benchmark_gen.py --testdir, path to the original test data, --savedir, path to save benchmark .labels file, --refdata, path to processed test txt file, --reflabel, path to processed test npy file or path new generated test npy file  `
I got a predicted .labels file and all the benchmark in this .labels file is 1.
Have you sloved your problem and cloud you share your experience?
Thank you.

**Comment by renwang435** [comment]
@aboulch If you have time, it would be great if you could lend us a hand in debugging this issue! I presume it's a simple matter of changing some paths around in the code, but I'm not sure which files should be modified, or how.

**Comment by aboulch** [comment]
Hello,
Very sorry for late response. For example let's say that 
- the benchmark test point cloud are in `/data/SEMANTIC3D/TEST`
- the pre-processed test point clouds are in `/data/SEMANTIC3D/test_voxel_0_1m/pointcloud_txt/`
- the predicted labels (for pre-processed point clouds) are in: `/data/SEMANTIC3D/ConvPoint/model/results/`
- the target directory is `/data/SEMANTIC3D/benchmark_results/`
Then the command line would be:

`python semantic3d_benchmark_gen.py --testdir /data/SEMANTIC3D/TEST/ --savedir /data/SEMANTIC3D/benchmark_results/ --refdata /data/SEMANTIC3D/test_voxel_0_1m/pointcloud_txt/ --reflabel /data/SEMANTIC3D/ConvPoint/model/results/`
